icu4c/source/common/unicode/platform.h

```diff
@@ -313,6 +313,20 @@
 #   define U_PLATFORM_IS_DARWIN_BASED 0
 #endif
 
+/**
+ * \def U_HAVE_ATOMICS
+ * Defines whether the platform supports atomic operations.
+ * @internal
+ */
+#ifdef U_HAVE_ATOMICS
+    /* Use the predefined value. */
+#elif defined(__wasi__) && !defined(_REENTRANT)
+    /* WASI does not support atomics when wasi-threads feature is not enabled */
```

Comment:
Question: why not use wasi-threads, then? It seems error-prone to add a flag that skips mutexes.